File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String winner : completion) {
            int count = map.getOrDefault(winner, 0);
            map.put(winner, count + 1);
        }
        
        for (String runner : participant) {
            int count = map.getOrDefault(runner, 0);
            if (count == 0) {
                return runner;
            }
            map.put(runner, count - 1);
        }
        
        return null;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    
    participant.forEach(runner => {
        const value = (map.get(runner) || 0) + 1;
        map.set(runner, value);
    });
    completion.forEach(runner => {
        const value = map.get(runner) - 1;
        map.set(runner, value);
    });
    
    for (const [runner, value] of map.entries()) {
        if (value > 0) {
            return runner;
        }
    }
    
    return 5;
}